### PR TITLE
Fix retry-provisioning command

### PR DESCRIPTION
**Pull request by wallyworld**
The juju retry-provisioning command was broken because it was updating the machine agent status, not the machine instance status when it set the "transient=true" status flag.
This PR fixes that, and also addresses a long standing issue where the facade code to do the update was in the wrong package. The update code is moved from the apiserver/common package to the client facade package.

## QA steps

Hard to QA unless the provider returns an error. I hard coded the LXD provider to return an error and added a machine. AFter the provisioning failed, I ran juju retry-provisioning and saw the operation being retried using juju status.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1938736

### CODE REVIEW

File: apiserver/facades/client/client/client.go [hunk 1]
```diff
@@ -876,13 +871,60 @@ func (c *Client) RetryProvisioning(p params.Entities) (params.ErrorResults, erro
...
+	}
+	return result, nil
+}
+
+type instanceStatus interface {
+	InstanceStatus() (status.StatusInfo, error)
+	SetInstanceStatus(sInfo status.StatusInfo) error
+}
+
+func (c *Client) updateInstanceStatus(tag names.Tag, data map[string]interface{}) error {
```
Comment (wallyworld) [inline apiserver/facades/client/client/client.go:895]: moved across from apiserver/common

**Comment by wallyworld** [comment]
[retry provisioning broken](https://trello.com/c/d8W5uqW4/1696-retry-provisioning-broken)

**Review by tlm — APPROVED** [review]
lgtm

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**